### PR TITLE
only liable(responsible) tenants are shown as joint tenants

### PR DESCRIPTION
**Pull request by elena-vi**
users need to know if the tenancy is a joint tenancy with all names on the tenancy so I can ensure that case details are disclosed to the correct person. Meaning the person(s) marked liable

https://hackney.atlassian.net/secure/RapidBoard.jspa?rapidView=22&projectKey=MA&modal=detail&selectedIssue=MA-69

### CODE REVIEW

File: LBHTenancyAPI/Gateways/V1/UhTenanciesGateway.cs [hunk 1]
```diff
@@ -171,7 +171,7 @@ public Tenancy GetTenancyForRef(string tenancyRef)
                     LEFT JOIN property WITH(NOLOCK)
                     ON property.prop_ref = tenagree.prop_ref
                     Left JOIN dbo.member member WITH(NOLOCK)
-                    ON member.house_ref = tenagree.house_ref
+                    ON member.house_ref = tenagree.house_ref and member.responsible = 1
```
Comment (bjpirt) [inline LBHTenancyAPI/Gateways/V1/UhTenanciesGateway.cs]: wouldn't this AND clause go in the WHERE conditions?

Comment (elena-vi) [reply LBHTenancyAPI/Gateways/V1/UhTenanciesGateway.cs]: yes

File: LBHTenancyAPITest/Test/Gateways/V1/UhTenanciesGatewayTest.cs [hunk 2]
```diff
@@ -122,10 +123,21 @@ private Tenancy GenerateTenancy(string tenency_ref = "")
             expectedTenancy.prop_ref = expectedProperty.prop_ref;
             expectedTenancy.start_date = expectedTenancy.start_date;
             TestDataHelper.InsertTenancy(expectedTenancy, _databaseFixture.Db);
-            //member 1
+
             var expectedMember = Fake.UniversalHousing.GenerateFakeMember();
-            expectedMember.house_ref = expectedTenancy.house_ref;
-            TestDataHelper.InsertMember(expectedMember, _databaseFixture.Db);
+            if (household != null && household.Any())
```
Comment (kjdchapman) [inline LBHTenancyAPITest/Test/Gateways/V1/UhTenanciesGatewayTest.cs]: Would it be better to default the household to an empty array and make this check simpler? Or do we expect the caller to pass `null` to us explicitly sometimes?

File: LBHTenancyAPITest/Test/Gateways/V1/UhTenanciesGatewayTest.cs [hunk 3]
```diff
@@ -379,11 +391,20 @@ public void WhenGivenTenancyRef_GetSingleTenancyByRef_ShouldReturnTenancyWithBas
...
+            var expectedTenant2 = Fake.UniversalHousing.GenerateFakeMember(responsible: true);
+
+            var householdMembers = new List<Member>()
+            {
+                expectedTenant, expectedTenant2,
+                Fake.UniversalHousing.GenerateFakeMember(responsible: false),
+                Fake.UniversalHousing.GenerateFakeMember(responsible: false)
+            };
+
+            Tenancy expectedTenancy = GenerateTenancy(household: householdMembers);
```
Comment (kjdchapman) [inline LBHTenancyAPITest/Test/Gateways/V1/UhTenanciesGatewayTest.cs]: Could we pull up the context of this test into the naming? I had to read it a couple of times to "get" it was four tenants, two of which are 'responsible', and only the responsible ones show up.

`public void GivenMixedTenants_GetSingleTenancyByRef_ReturnsOnlyResponsibleTenants()`?

Still a bit of a monster! Not sure how to do this without context blocks, not familiar with the language/framework.

**Pull request merged**